src/com/softwaret/kpdf/response/file/FileResponse.kt

```diff
@@ -0,0 +1,38 @@
+package com.softwaret.kpdf.response.file
+
+import io.ktor.http.*
+
+data class FileResponse(
+    val code: HttpStatusCode,
+    val fileName: String,
+    val file: ByteArray
+) {
+
+    override fun equals(other: Any?): Boolean {
```

Comment:
Dlaczego nie polegamy na normalnym equality data class?

Reply:
Bo jest `ByteArray` i nawet IDE zaleca nadpisanie tego i nie branie tego pola pod uwagę. Jak sobie zdekompilujesz `ByteCode` to widać normalną implementację 
```Java
  public boolean equals(@Nullable Object var1) {
      if (this != var1) {
         if (var1 instanceof FileResponse) {
            FileResponse var2 = (FileResponse)var1;
            if (Intrinsics.areEqual(this.code, var2.code) && Intrinsics.areEqual(this.fileName, var2.fileName) && Intrinsics.areEqual(this.file, var2.file)) {
               return true;
            }
         }

         return false;
      } else {
         return true;
      }
   }
```

Reply:
Aka chodzi o domyślny sposób porównywania `array`'ów w defaultowej impl `data class`